alt_e2eshark/e2e_testing/backends.py

```diff
@@ -94,6 +96,55 @@ def func(x):
 
         return func
 
+class CLIREEBackend(BackendBase):
+    '''This backend calls iree through the command line to compile and run MLIR modules'''
+    def __init__(self, *, device="local-task", hal_target_backend="llvm-cpu", extra_args : List[str] = None):
+        self.device = device
+        self.hal_target_backend = hal_target_backend
+        self.extra_args = []
+        if extra_args:
+            for a in extra_args:
+                if a[0:2] == "--":
+                    self.extra_args.append(a)
+                else:
+                    self.extra_args.append("--" + a)
+    
+    def compile(self, module_path: str, *, save_to : str = None) -> str:
+        vmfb_path = save_to + "compiled_model.vmfb"
+        arg_string = f"--iree-hal-target-backends={self.hal_target_backend} "
+        for arg in self.extra_args:
+            arg_string += arg
+            arg_string += " "
+        command_error_dump = os.path.join(save_to, "detail/compilation.detail.log")
+        commands_log = os.path.join(save_to, "commands/compilation.commands.log")
```

Comment:
change to os.path.join for the whole thing instead of unix style paths here -- just for windows compatibility

Reply:
You can also use `pathlib.Path` (https://docs.python.org/3/library/pathlib.html). I find that much nicer to use than `os` for path manipulation.

Reply:
Yeah, I really like pathlib, personally. I have half a mind to convert all of this testsuite to using `pathlib.Path` instead of `str`, but sometimes I do actually need to have a string, and I start with a string, so `os.path.join` seems easier than `str(Path(thing).joinpath(otherthing))`.

Reply:
I wonder if it's doubly bad that I use pathlib *sometimes*. Especially when getting parent directories, names, etc.